fgpyo/sequence.py

```diff
@@ -70,9 +71,19 @@ def reverse_complement(bases: str) -> str:
     return "".join([_COMPLEMENTS[b] for b in bases[::-1]])
 
 
-def longest_hp_length(bases: str) -> int:
+def gc_content(bases: str) -> float:
+    """Calculates the fraction of G and C bases in a sequence."""
+    if len(bases) == 0:
+        return 0
+    gc_count = sum(1 for base in bases if base == "C" or base == "G" or base == "c" or base == "g")
```

Comment:
I've updated my suggestions above to use `base in "CG"` instead of `base in ["C", "G"]`

`base in "CG"` will be about twice as performant as `base in ["C", "G"]`, and roughly as performant as `base in {"C", "G"}` while being (imo) more readable

```py
In [1]: %timeit "G" in "CG"
12.5 ns ± 0.0861 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)

In [2]: %timeit "G" in ["C", "G"]
19.8 ns ± 0.136 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [3]: %timeit "G" in {"C", "G"}
11.2 ns ± 0.143 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```

Reply:
@msto You didn't include a timing for multiple equality checks ... so I took a look myself, and at least for me, multiple equality checks is 3-4 times _slower_ than using `ch in "ACGT"`.  So +1 for using `in` over multiple equality checks.

Reply:
I think however, that calling `.upper()` first is probably less desirable since this might get called on long sequences or called frequently, and checking `in "CGcg"` is probably faster than calling `.upper()` on all inputs?